waspc/packages/deploy/src/providers/fly/setup/setup.ts

```diff
@@ -43,37 +43,74 @@ export async function setup(baseName: string, region: string, options: CommonOpt
 	waspSays(`Don't forget to create your database by running "${getCommandHelp(createFlyDbCommand)}".`);
 }
 
-async function setupServer(deploymentInfo: DeploymentInfo) {
+async function setupServer(deploymentInfo: DeploymentInfo<SetupOptions>) {
 	waspSays(`Setting up server app with name ${deploymentInfo.serverName}`);
 
 	cdToServerBuildDir(deploymentInfo.options.waspProjectDir);
 	deleteLocalToml();
 
+	const launchArgs = [
+		'--name', deploymentInfo.serverName,
+		'--region', deploymentInfo.region,
+	];
+
+	if (deploymentInfo.options.org) {
+		launchArgs.push('--org', deploymentInfo.options.org);
+	}
+
 	// This creates the fly.toml file, but does not attempt to deploy.
-	await $`flyctl launch --no-deploy --name ${deploymentInfo.serverName} --region ${deploymentInfo.region}`;
+	await $`flyctl launch --no-deploy ${launchArgs}`;
 
 	copyLocalServerTomlToProject(deploymentInfo.tomlFilePaths);
 
 	const randomString = crypto.randomBytes(32).toString('hex');
-	await $`flyctl secrets set JWT_SECRET=${randomString} PORT=8080 WASP_WEB_CLIENT_URL=${deploymentInfo.clientUrl}`;
+
+	const secretsArgs = [
+		`JWT_SECRET=${randomString}`,
+		// NOTE: Normally these would just be envars, but flyctl
+		// doesn't provide a way to set envars that persist to fly.toml.
+		'PORT=8080',
+		`WASP_WEB_CLIENT_URL=${deploymentInfo.clientUrl}`,
+	];
+
+	if (deploymentInfo.options.serverSecret.length > 0) {
+		deploymentInfo.options.serverSecret.forEach(secret => {
+			secretsArgs.push(secret);
```

Comment:
I see that you are passing it in as-is (strings in, strings passed to Fly CLI), maybe consider parsing the user input into `Secret` type to make sure that they provided valid secrets and then we stringily it here again.

Reply:
Can be chewed on more here: https://github.com/wasp-lang/wasp/pull/1196#discussion_r1196473220